antenna/external/fs/crashstorage.py

```diff
@@ -0,0 +1,132 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+import os
+import os.path
+
+from everett.component import ConfigOptions, RequiredConfigMixin
+
+from antenna.util import json_ordered_dumps
+
+
+logger = logging.getLogger(__name__)
+
+
+class FSCrashStorage(RequiredConfigMixin):
+    """Saves raw crash files to the file system
+
+    This generates a tree something like this which mirrors what we do
+    on S3:
+
+    ::
+
+        <FS_ROOT>/
+            <YYYYMMDD>/
+                raw_crash/
+                    <CRASHID>.json
+                dump_names/
+                    <CRASHID>.json
+                <DUMP_NAME>/
+                    <CRASHID>
+
+
+    Couple of things to note:
+
+    1. This doesn't ever delete anything from the tree. You should run another
+       process to clean things up.
+
+    2. If you run out of disk space, this component will fail miserably.
+       There's no way to recover from a full disk--you will lose crashes.
+
+    FIXME(willkg): Can we alleviate or reduce the likelihood of the above?
+
+    """
+    required_config = ConfigOptions()
+    required_config.add_option(
+        'fs_root',
+        default='/tmp/antenna_crashes',
+        doc='path to where files should be stored'
+    )
+
+    # FIXME(willkg): umask
```

Comment:
The `FSTemporaryStorage` has a umask configuration option, but I didn't add one here. Having one is probably a good idea if we ever "went production" with this.